clio-client/src/main/scala/org/broadinstitute/clio/client/commands/MoveWgsUbamCommand.scala

```diff
@@ -31,63 +28,96 @@ object MoveWgsUbamCommand
-      _ <- deleteGoogleObject(wgsUbam.ubamPath, ioUtil) withErrorMsg
+      _ <- deleteGoogleObject(wgsUbamPath, ioUtil) withErrorMsg
         """The old bam was not able to be deleted. Clio has been updated to point to the new bam.
-          | Please delete the old bam. If this cannot be done, contact Green Team at greenteam@broadinstitute.org
+          | Please delete the old bam. If this cannot be done, contact Green Team at greenteam@broadinstitute.org.
           | """.stripMargin
     } yield {
+      logger.info(
+        s"Successfully moved '${wgsUbamPath.get}' to '${config.ubamPath.get}'"
+      )
       upsertUbam
     }
   }
 
-  private def queryForWgsUbam(
-    webClient: ClioWebClient,
-    config: BaseArgs
-  ): Future[TransferWgsUbamV1QueryOutput] = {
-    val httpResponse = webClient.queryWgsUbam(
-      config.bearerToken.getOrElse(""),
-      TransferWgsUbamV1QueryInput(
-        flowcellBarcode = config.flowcell,
-        lane = config.lane,
-        libraryName = config.libraryName,
-        location = Option(Location.GCP),
-        documentStatus = Option(DocumentStatus.Normal)
-      )
-    )
+  private def verifyPath(webClient: ClioWebClient,
+                         config: BaseArgs): Future[Unit] = {
+    implicit val executionContext: ExecutionContext = webClient.executionContext
+    Future(config.location.foreach {
+      case "GCP" => ()
+      case _ =>
+        logger.error("Only GCP unmapped bams are supported at this time.")
+        throw new Exception()
+    }).flatMap { _ =>
+      Future(config.ubamPath.foreach {
+        case loc if loc.startsWith("gs://") => ()
+        case _ =>
+          logger.error(
+            s"The destination of the ubam must be a cloud path. ${config.ubamPath.get} is not a cloud path."
+          )
+          throw new Exception()
+      })
+    }
+  }
+
+  private def queryForWgsUbamPath(webClient: ClioWebClient,
+                                  config: BaseArgs): Future[Option[String]] = {
     implicit val ec: ExecutionContext = webClient.executionContext
+
+    def ensureOnlyOneJson(wgsUbams: Json): Json = {
+      wgsUbams.asArray.getOrElse(throw new Exception()).size match {
+        case 1 =>
+          wgsUbams.asArray.get.head
+        case s if s > 1 =>
+          logger.error(
+            s"$s WgsUbams were returned for Key(${prettyKey(config)}), expected 1."
+          )
+          throw new Exception()
+        case s if s == 0 =>
+          logger.error(s"No WgsUbams were found for Key(${prettyKey(config)})")
+          logger.error("You can add this WgsUbam using the AddWgsUbam command in the Clio client.")
+          throw new Exception()
+      }
+    }
+
+    // Using JSON instead of TransferWgsUbamV1QueryOutput here on purpose.
+    // Currently, the clio server spits json out in snake_case.
+    // Clio client expects it json keys to be in camelCase.
+    // This should be changed to use the transfer model once the issue is fixed
```

Comment:
Is there a JIRA for this issue? This seems like something we should iron out early on.

Reply:
I'd rather we fix it now. The snake-case config is defined in `ModelAutoDerivation`, so it should be applying. @tlangs are you sure that was the cause of whatever error led you to do this?

Reply:
I assumed that was the problem. It was having errors with fields like 'flowcell_barcode', but when I tried some custom data with 'flowcellBarcode', it worked fine...

Reply:
Let's huddle about dis.